src/sentry/nodestore/base.py

```diff
@@ -58,39 +65,120 @@ def delete_multi(self, id_list):
         for id in id_list:
             self.delete(id)
 
-    def get(self, id):
+    def _decode(self, value, subkey):
+        if value is None:
+            return None
+
+        lines_iter = iter(value.splitlines())
+        try:
+            if subkey is not None:
+                next(lines_iter)
+
+                for line in lines_iter:
+                    if line.strip() == subkey.encode("utf8"):
```

Comment:
The `splitlines` iter and repeated `encode` can be optimized. 

Also, you can skip comparing every second line.

Reply:
encode yes, splitlines is implemented in C so I can't beat it (and have to suck up memory cost I guess)

found some stats online: https://stackoverflow.com/a/3054898/1544347